tidy3d/freq_range.py

```diff
@@ -0,0 +1,237 @@
+
+            Returns
+            -------
+            updated instance of class FreqRange
+
+            Example
+            -------
+            >>> wvl_min = 1e-7
+            >>> wvl_max = 1e-6
+            >>> freq_range = FreqRange.from_wavelength(wvl_min, wvl_max)
+        """
+
+        if wvl_max <= wvl_min:
+            raise ValueError("longest wavelength must be greater than the shortest wavelength")
+        if wvl_min <= 0:
+            raise ValueError("wavelength has to be a positive value")
+
+        # convert wavelength intervals to freq. range
+        lda0  = 0.5 * (wvl_min + wvl_max)     # get central wavelength
+        freq0 = um_to_m(td_const.C_0 / lda0)  # get central frequency
+        fmax = um_to_m(td_const.C_0 / wvl_min)
+        fmin = um_to_m(td_const.C_0 / wvl_max)
+        fwidth = 0.5 * (fmax - fmin)     # define width of source frequency range
+
+        
+        freq_range = clf(freq0, fwidth)
+
+        freq_range.fmin = fmin
+        freq_range.fmax = fmax
+        return freq_range
+    
+
+    def samples(self, num_points):
+        """
+            method samples() performes discretization of the frequency range by a
+            given number of points "num_points"; frequency samples are uniformly 
+            distributed in ascending order (e.g. freqs = [3,4,...,100]Hz)
+
+            Parameters
+            ----------
+            num_points - number of frequency points 
+
+            Returns
+            -------
+            freqs - a numpy array of uniformly distributed frequency samples in ascending order
+        """
+
+        # update number of freq points 
+        self.num_points = num_points
+
+        if num_points == 1:     # if one sample frequency point is selected
+
+            self.freqs = np.array([self.freq0])
+            self.ldas  = np.array([self.lda0])
+
+        else:                   # otherwise 
+
+            if self.fmax <= self.fmin:
+                raise ValueError("fmin has to be larger than fmin")
+            if self.fmin <=0:
+                raise ValueError("frequencies have to be positive values")
+
+            # calculate frequency points and corresponding wavelengths
+            self.freqs = np.linspace(self.fmin, self.fmax, self.num_points)
+
+            # define shortest and longest wavelengths
+            lmin = um_to_m(td_const.C_0 / self.fmax)
+            lmax = um_to_m(td_const.C_0 / self.fmin) 
+
+            # update array of wavelengths (wavelengths would be in descending order) 
+            self.ldas  = um_to_m(td_const.C_0 / self.freqs)
+
+        return self.freqs
+    
+    def gaussian_pulse(self):
+        """
+            method gaussian_pulse() returns instance of class GaussianPulse
+            with central frequency freq0 and width of the source frequency fwidth
+        """
+
+        # create an instance of GaussianPulse class with defined frequency params
+        return GaussianPulse(freq0 = self.freq0, fwidth = self.fwidth)
```

Comment:
Better to use [from_frequency_range](https://github.com/flexcompute/tidy3d/blob/16607e9985e7ba4664ed5330368f1e7a911be622/tidy3d/components/source/time.py#L189) here because this method ensures a good amount of power in the pulse signal across the frequency range.